topo/node/magna/magna.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package magna implements the node definition for a magna container in KNE.
+package magna
+
+import (
+	"fmt"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+	"github.com/openconfig/kne/topo/node"
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	cfg := defaults(nodeImpl.Proto)
+	nodeImpl.Proto = cfg
+	n := &Node{
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+func defaults(pb *tpb.Node) *tpb.Node {
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	if len(pb.GetConfig().GetCommand()) == 0 {
+		pb.Config.Command = []string{
+			"/app/magna",
+			"-v=2",
+			"-alsologtostderr",
+			"-port=40051",
```

Comment:
are these meant to be default services and self-signed certs? if so please add a TODO to expose those as such otherwise ondatra will not be able to resolve the gnmi port as its not encoded in the textproto unless expressed as a service

Reply:
ah, no problem -- yes, they're two services that are exposed. i'll just go and make these default services in this implementation.

Reply:
done.